From 7fc69e0190e08fd809824d5d2da649c9082ebde1 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:36 -0400
Subject: [fs] btrfs: send, remove dead code from __get_cur_name_and_parent

Message-id: <1407179109-13714-330-git-send-email-sandeen@redhat.com>
Patchwork-id: 87231
O-Subject: [RHEL7.1 PATCH 329/422] Btrfs: send, remove dead code from __get_cur_name_and_parent
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Filipe Manana <fdmanana@gmail.com>

commit a10c40766c30a002affc0c47dd515d048c3959b4
Author: Filipe Manana <fdmanana@gmail.com>
Date:   Sat Mar 22 17:16:30 2014 +0000

    Btrfs: send, remove dead code from __get_cur_name_and_parent

    Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/send.c b/fs/btrfs/send.c
index c2bfd60..fb6aeed 100644
--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -2030,7 +2030,6 @@ static int __get_cur_name_and_parent(struct send_ctx *sctx,
 {
  int ret;
  int nce_ret;
- struct btrfs_path *path = NULL;
  struct name_cache_entry *nce = NULL;
 
  /*
@@ -2056,10 +2055,6 @@ static int __get_cur_name_and_parent(struct send_ctx *sctx,
   }
  }
 
- path = alloc_path_for_send();
- if (!path)
-  return -ENOMEM;
-
  /*
   * If the inode is not existent yet, add the orphan name and return 1.
   * This should only happen for the parent dir that we determine in
@@ -2135,7 +2130,6 @@ out_cache:
  name_cache_clean_unused(sctx);
 
 out:
- btrfs_free_path(path);
  return ret;
 }
 
-- 
1.7.1